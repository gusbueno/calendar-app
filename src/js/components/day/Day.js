import React from 'react'
import PropTypes from 'prop-types'

import styles from './Day.scss'

const renderEvents = (events, onUpdateEvent) => {
  return events.map(event => {
    return (
      <li className={styles.event} key={event.id} onClick={() => onUpdateEvent(event)}>{event.title}</li>
    )
  })
}

const Day = ({ dayNumber, onAddEvent, tsDate, events, onUpdateEvent }) => {
  if (!dayNumber) {
    return <div className={styles.day} />
  }

  return (
    <div className={styles.day}>
      <div className={styles['day-header']}>
        <span>{dayNumber}</span>
        <div className={styles['add-event-btn']} onClick={() => onAddEvent({ ts: tsDate })}>
          <span>+</span>
        </div>
      </div>
      <div className={styles['day-body']}>
        <ul className={styles.events}>
          {renderEvents(events, onUpdateEvent)}
        </ul>
      </div>
    </div>
  )
}

Day.propTypes = {
  dayNumber: PropTypes.number,
  onAddEvent: PropTypes.func,
  tsDate: PropTypes.number,
  events: PropTypes.array,
  onUpdateEvent: PropTypes.func
}

Day.defaultProps = {
  onAddEvent: () => {},
  onUpdateEvent: () => {}
}

export default Day
