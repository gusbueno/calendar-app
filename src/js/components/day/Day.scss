@import "../../../styles/variables";

.day {
  padding: 5px;
  box-sizing: border-box;
  border: 1px solid $color-green;
  width: 180px;
  height: 150px;
  display: flex;
  flex-direction: column;

  .day-header {
    display: flex;
    justify-content: space-between;
    align-items: center;

    >span {
      font-size: 12px;
      font-family: 'Roboto-Bold';
    }

    .add-event-btn {
      width: 20px;
      height: 20px;
      border-radius: 50%;
      background-color: $color-green;
      display: flex;
      justify-content: center;
      align-items: center;
      &:hover {
        cursor: pointer;
      }

      span {
        color: $color-white;
        font-size: 14px;
        font-family: 'Roboto-Light';
        text-align: center;
      }
    }
  }
}