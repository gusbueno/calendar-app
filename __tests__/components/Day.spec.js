/* global describe, it, expect, jest */

import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import Day from '../../src/js/components/day/Day'

describe('<Day /> Component', () => {
  it('should match with the snapshot', () => {
    const tree = renderer.create(
      <Day dayNumber={1} onAddEvent={() => {}} tsDate={601293600} />
    )
    const json = tree.toJSON()
    expect(json).toMatchSnapshot()
  })

  const onAddEventMock = jest.fn()
  const dayShallow = shallow(
    <Day dayNumber={1} onAddEvent={onAddEventMock} tsDate={601293600} />
  )

  it('should exists', () => {
    expect(dayShallow.length).toEqual(1)
  })

  it('should render day div', () => {
    expect(dayShallow.find('.day').length).toEqual(1)
  })

  it('should render day-header div', () => {
    expect(dayShallow.find('.day-header').length).toEqual(1)
  })

  it('should render add-event-btn div', () => {
    expect(dayShallow.find('.add-event-btn').length).toEqual(1)
  })

  it('should call onAddEvent with tsDate value on click add-event-btn', () => {
    dayShallow.find('.add-event-btn').first().simulate('click')
    expect(onAddEventMock).toHaveBeenCalledWith(601293600)
  })

  /* empty Day */
  const emptyDayShallow = shallow(
    <Day />
  )

  it('should render day div on empty day', () => {
    expect(emptyDayShallow.find('.day').length).toEqual(1)
  })

  it('should day div have no children', () => {
    expect(emptyDayShallow.find('.day').children()).toHaveLength(0)
  })
})
