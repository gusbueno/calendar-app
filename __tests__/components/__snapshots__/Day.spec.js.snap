// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Day /> Component should match with the snapshot 1`] = `
<div
  className="day"
>
  <div
    className="day-header"
  >
    <span>
      1
    </span>
    <div
      className="add-event-btn"
      onClick={[Function]}
    >
      <span>
        +
      </span>
    </div>
  </div>
</div>
`;
